r/R/dplyr-funcs-datetime.R

```diff
@@ -300,6 +312,34 @@ register_bindings_duration <- function() {
 
     build_expr("cast", x, options = cast_options(to_type = duration(unit = "s")))
   })
+  register_binding("decimal_date", function(date) {
+    # browser()
+    y <- build_expr("year", date)
+    # timezone <- call_binding("tz", date)
+    start <- call_binding("make_datetime", year = y, tz = "UTC")
+    end <- call_binding("make_datetime", year = y + 1L, tz = "UTC")
+    # maybe use yday here
+    sofar <- call_binding("difftime", date, start, units = "secs")
+    total <- call_binding("difftime", end, start, units = "secs")
+    y + sofar$cast(int64()) / total$cast(int64())
+  })
+  register_binding("date_decimal", function(decimal, tz = "UTC") {
+    y <- build_expr(
+      "cast",
+      decimal,
+      options = cast_options(to_type = int32(), safe = FALSE)
+    )
+
+    start <- call_binding("make_datetime", year = y, tz = tz)
+    end <- call_binding("make_datetime", year = y + 1, tz = tz)
+
+    seconds <- call_binding("difftime", end, start, units = "secs")
```

Comment:
`seconds` here is effectively the number of seconds in a year, yeah? Would it be better or worse to have 31536000 (or 31622400 for leap years) instead of calculating and constructing it here?

Reply:
done